nat-lab/tests/utils/connection_tracker.py

```diff
@@ -102,23 +114,33 @@ async def on_stdout(self, stdout: str) -> None:
         if not self._config:
             return
 
-        for line in stdout.splitlines():
-            connection = parse_input(line)
-            if connection is FiveTuple():
-                continue
+        with self._lock:
+            for line in stdout.splitlines():
+                connection = parse_input(line)
+                if connection is FiveTuple():
+                    continue
+
+                if not self._initialized:
```

Comment:
What is the point of this, if we are now have full state object

Reply:
I kept the old initialize mechanism in, but we could also re-use synchronize() for it.